src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -360,6 +398,45 @@ class WelcomePage extends Disposable {
 				windowOpenable = { workspaceUri: recent.workspace.configPath };
 			}
 
+			const daysOfWeek: string[] = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+			const monthsOfYear: string[] = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'];
+
+			let lastOpened: string;
+			let mtime: any = fs.statSync(fullPath).mtime;
+			let currentDate: Date = new Date;
+			let daysLapsed: number;
+			let delta = Math.abs(mtime - currentDate.getTime()) / 1000;
+
+			let hours = mtime.getHours();
+			let minutes = mtime.getMinutes();
+			let ampm = hours >= 12 ? 'pm' : 'am';
+			hours = hours % 12;
+			hours = hours ? hours : 12; // the hour '0' should be '12'
+			minutes = minutes < 10 ? '0' + minutes : minutes;
+			let strTime = hours + ':' + minutes + ' ' + ampm;
+
+			// calculate (and subtract) whole days
+			daysLapsed = Math.floor(delta / 86400);
+			delta -= daysLapsed * 86400;
+
+			// what's left is seconds
+			let secondsLapsed = delta % 60;  // in theory the modulus is not required
+
+			if (daysLapsed === 0 && secondsLapsed < 60) {
+				lastOpened = `Just Now`;
+			}
+			else if (daysLapsed === 1) {
+				lastOpened = `Yesterday at ${strTime}`;
+
+			}
+			else if (daysLapsed <= 7) {
```

Comment:
as @Charles-Gagnon mentions, this whole date-related code isn't localization\globalization friendly.

Reply:
localized